tensorflow/core/kernels/svd_op_gpu.cu.cc

```diff
@@ -0,0 +1,414 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// See docs in ../ops/linalg_ops.cc.
+// TODO(shamanDevel): Enable complex inputs. This will require additional tests
+//                    and OP_REQUIRES.
+#if GOOGLE_CUDA
+#define EIGEN_USE_GPU
+
+#include <algorithm>
```

Comment:
do you still need <algorithm> ?

Reply:
Do you mean EIGEN_USE_GPU or <algorithm>? I need both. EIGEN_USE_GPU for the GPU_Device, and algorithm for p=std::min(m, n)